### PR TITLE
Add input box for per page #2427

### LINKED ISSUE #2427: IMPORTER: Investigate ways to increase speed of the Importer
## Issue Overview
When importing 1000+ donations, the importer timed out locally. Importing 500+ donations took around 30 minutes. There should be smart ways to increase the speed of the imports, or even set an amount of donations to import at a time.

I have a couple CSV files available for testing purposes.

### PR DESCRIPTION
**Pull request by raftaar1191**
## Description
PR to resolved #2427 

## How Has This Been Tested?
By uploading the donation and update the per page section input box

See the video for more information:http://screencast-o-matic.com/watch/cbXvXj24vO

**Notes: Here the `per page` is set to 25** 

## Screenshots (jpeg or gifs if applicable):
image

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/admin/tools/import/class-give-import-donations.php [hunk 1]
```diff
@@ -57,9 +57,9 @@
 		 * @access private
 		 */
 		private function __construct() {
+			self::$per_page  = absint( ! empty( $_REQUEST['per_page'] ) ? $_REQUEST['per_page'] : self::$per_page );
```
Comment (ravinderk) [inline includes/admin/tools/import/class-give-import-donations.php]: @raftaar1191 Using `$_REQUEST` is bad practice, please use appropriate HTTP method

Comment (raftaar1191) [reply includes/admin/tools/import/class-give-import-donations.php]: Thanks @ravinderk for pointing this out

Comment (raftaar1191) [reply includes/admin/tools/import/class-give-import-donations.php]: Fixed

File: includes/admin/tools/import/class-give-import-donations.php [hunk 2]
```diff
@@ -57,9 +57,9 @@
 		 * @access private
 		 */
 		private function __construct() {
+			self::$per_page  = absint( ! empty( $_GET['per_page'] ) ? $_GET['per_page'] : self::$per_page );
```
Comment (ravinderk) [inline includes/admin/tools/import/class-give-import-donations.php]: @raftaar1191 Can you explain `absint( ! empty( $_GET['per_page'] )`

File: includes/admin/tools/import/class-give-import-donations.php [hunk 3]
```diff
@@ -724,6 +724,7 @@ public function render_media_csv() {
 			if ( empty( $csv_id ) || ! $this->is_valid_csv( $csv_id, $csv ) ) {
 				$csv_id = $csv = '';
 			}
+			$per_page = absint( isset( $_POST['per_page'] ) ? give_clean( $_POST['per_page'] ) : self::$per_page );
```
Comment (ravinderk) [inline includes/admin/tools/import/class-give-import-donations.php]: @raftaar1191  can you explain `absint( isset( $_POST['per_page'] )`

File: includes/admin/tools/import/class-give-import-donations.php [hunk 4]
```diff
@@ -820,6 +828,7 @@ public function save() {
 						'delete_csv'    => empty( $_POST['delete_csv'] ) ?
 							'1' :
 							( give_is_setting_enabled( give_clean( $_POST['delete_csv'] ) ) ? '1' : '0' ),
+						'per_page'      => absint( isset( $_POST['per_page'] ) ? $_POST['per_page'] : self::$per_page ),
```
Comment (ravinderk) [inline includes/admin/tools/import/class-give-import-donations.php]: @raftaar1191 Can you explain `absint( isset( $_POST['per_page'] )`

File: includes/admin/tools/import/class-give-import-donations.php (hunk 2 above)
Comment (raftaar1191) [reply includes/admin/tools/import/class-give-import-donations.php]: Got it @ravinder
What you are pointing 
self::$per_page  = absint( ! empty( $_GET['per_page'] ) ? $_GET['per_page'] : self::$per_page );

I am using absint every time without depending on the situation 
Like if  $_GET['per_page'] is empty then I do not need to check self::$per_page as it datatype will always be INT

Comment (raftaar1191) [reply includes/admin/tools/import/class-give-import-donations.php]: Fixing that

Comment (raftaar1191) [reply includes/admin/tools/import/class-give-import-donations.php]: Update the code 

Thanks @ravinderk for pointing this out to me.

File: includes/admin/tools/import/class-give-import-donations.php (hunk 3 above)
Comment (raftaar1191) [reply includes/admin/tools/import/class-give-import-donations.php]: Update the code 

Thanks @ravinderk for pointing this out to me.

File: includes/admin/tools/import/class-give-import-donations.php (hunk 4 above)
Comment (raftaar1191) [reply includes/admin/tools/import/class-give-import-donations.php]: Update the code 

Thanks @ravinderk for pointing this out to me.

**Pull request merged**